am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -136,83 +137,72 @@ public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata accessMe
         }
 
         JsonNode filteredJson = filterService.filterJson(resource.getResourceJson(), attributePermissions);
 
-        return FilterResourceResponse.builder()
+        return FilteredResourceEnvelope.builder()
             .resourceId(resource.getResourceId())
+            .resourceDefinition(resource.getType())
             .data(filteredJson)
-            .permissions(attributePermissions)
+            .access(AccessEnvelope.builder()
+                .permissions(attributePermissions)
+                .accessManagementType(accessManagementType)
+                .build())
             .build();
     }
 
-    private boolean explicitAccessType(Set<String> userRoles) {
-        return jdbi.withExtension(AccessManagementRepository.class,
-            dao -> dao.getRoleAccessType(userRoles.iterator().next())).equals(AccessType.EXPLICIT);
+    private boolean roleBasedAccessType(String userRole) {
+        AccessType accessType = jdbi.withExtension(AccessManagementRepository.class,
+            dao -> dao.getRoleAccessType(userRole));
+
+        return accessType != null && accessType.equals(AccessType.ROLE_BASED);
     }
 
     /**
      * Retrieves a list of {@link RoleBasedAccessRecord } and returns attribute and permissions values.
      *
-     * @param serviceName  name of service
-     * @param resourceType type of resource
-     * @param resourceName name of a resource
-     * @param userRoles    A set of user roles
+     * @param resource  {@link ResourceDefinition} a unique service name, resource type and resource name
+     * @param userRoles A set of user roles
      * @return a map of attributes and their corresponding permissions or null
      * @throws PersistenceException if any persistence errors were encountered
      */
-    @AuditLog("returned role access to resource defined as '{{serviceName}}|{{resourceType}}|{{resourceName}}' "
+    @AuditLog("returned role access to resource "
+        + "defined as '{{resource.serviceName}}|{{resource.resourceType}}|{{resource.resourceName}}' "
         + "for roles '{{userRoles}}': {{result}}")
-    public Map<JsonPointer, Set<Permission>> getRolePermissions(@NotBlank String serviceName,
-                                                                @NotBlank String resourceType,
-                                                                @NotBlank String resourceName,
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(@NotNull @Valid ResourceDefinition resource,
                                                                 @NotEmpty Set<@NotBlank String> userRoles) {
         List<Map<JsonPointer, Set<Permission>>> permissionsForRoles =
             jdbi.withExtension(AccessManagementRepository.class, dao ->
                 userRoles.stream()
-                    .map(role -> dao.getRolePermissions(serviceName, resourceType, resourceName, role))
+                    .filter(this::roleBasedAccessType)
```

Comment:
Obviously that needs to be removed from the method logic since `getRolePermissions` should not reject any access type.

As a solution I'd suggest reducing list of user roles passed to `filterResource` method to contain only roles with role based access before such list be passed into `getRolePermissions` method.

Reply:
In case `getRolePermissions` is called by `filterResource` (both public methods) two audit logs will be generated. It is obviously unwanted behaviour that we need to find work around for. Solution would be either:
* extracting private method that is called by both `getRolePermissions` and `filterResource`
* trying to reconfigure auditing aspect to kick off only when public method wasn't called by other public method from the same service

Reply:
I was leaning towards extracting common logic to a private method called by both 👍